src/software/ai/navigator/path_planner/theta_star_path_planner.cpp

```diff
@@ -174,123 +171,145 @@ Path ThetaStarPathPlanner::findPath(const Point &start, const Point &destination
                                     const Rectangle &navigable_area,
                                     const std::vector<Obstacle> &obstacles)
 {
-    obstacles_ = obstacles;
-    Path empty_ret_val(std::nullopt);
-    CellCoordinate src_coord, dest_coord;
-
-    openList.clear();
+    // Initialize member variables
+    this->obstacles       = obstacles;
+    max_navigable_x_coord = navigable_area.xLength() / 2.0 - ROBOT_MAX_RADIUS_METERS;
+    max_navigable_y_coord = navigable_area.yLength() / 2.0 - ROBOT_MAX_RADIUS_METERS;
+    num_grid_rows = (int)((max_navigable_x_coord * 2.0 + ROBOT_MAX_RADIUS_METERS) /
+                          SIZE_OF_GRID_CELL_IN_METERS);
+    num_grid_cols = (int)((max_navigable_y_coord * 2.0 + ROBOT_MAX_RADIUS_METERS) /
+                          SIZE_OF_GRID_CELL_IN_METERS);
+
+    // Reset data structures to path plan again
+    open_list.clear();
+    closed_list.clear();
     unblocked_grid.clear();
 
-    fieldXLength     = navigable_area.xLength();
-    fieldYLength     = navigable_area.yLength();
-    fieldXHalfLength = fieldXLength / 2.0;
-    fieldYHalfLength = fieldYLength / 2.0;
+    // Initialize local variables
+    Point closest_destination = findClosestFreePoint(destination);
+    Coordinate src_coord      = pointToCoordinate(start);
+    Coordinate dest_coord     = pointToCoordinate(closest_destination);
 
-    // initialize grid
-    numRows =
-        (int)((fieldXLength - ROBOT_MAX_RADIUS_METERS) / SIZE_OF_GRID_CELL_IN_METERS);
-    numCols =
-        (int)((fieldYLength - ROBOT_MAX_RADIUS_METERS) / SIZE_OF_GRID_CELL_IN_METERS);
+    bool isInvalidOrBlocked = checkForInvalidOrBlockedCases(src_coord, dest_coord);
+    if (isInvalidOrBlocked)
+    {
+        return std::nullopt;
+    }
 
-    Point closest_destination = findClosestFreePoint(destination);
-    src_coord                 = convertPointToCell(start);
-    dest_coord                = convertPointToCell(closest_destination);
-    // If the source is out of range
-    if (isValid(src_coord.first, src_coord.second) == false)
+    if (isStartToDestinationWithinThreshold(start, destination))
     {
-        LOG(WARNING) << "Source is not valid; no path found" << std::endl;
-        return empty_ret_val;
+        return Path(std::vector<Point>({start, destination}));
     }
 
-    // If the destination is out of range
-    if (isValid(dest_coord.first, dest_coord.second) == false)
+    if (isStartToClosestDestinationWithinThreshold(start, closest_destination))
     {
-        LOG(WARNING) << "Destination is not valid; no path found" << std::endl;
-        return empty_ret_val;
+        return Path(std::vector<Point>({start, closest_destination}));
     }
 
-    if ((start - destination).len() < CLOSE_TO_DEST_THRESHOLD ||
-        ((start - destination).len() < SIZE_OF_GRID_CELL_IN_METERS))
+    initListsAndCellDetails(src_coord);
+
+    bool found_dest = findPathToDestination(dest_coord);
+
+    if (found_dest == false)
```

Comment:
This is only used once?

Reply:
I rather have the function call outside the if statement because findPathToDestination does some work